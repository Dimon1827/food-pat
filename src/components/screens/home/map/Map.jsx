import Contacts from './contacts/Contacts';
import './Map.scss';

function Map() {
  return (
    <section className="map">
      <div className="container">
        <h2 className="visually-hidden">Контакты</h2>
        <Contacts />
      </div>
      <picture>
        <source
          type="image/webp"
          media="(min-width:1280px)"
          srcSet="
                images/location-desktop@1x.webp,
                images/location-desktop@2x.webp 2x
              "
          width="1440"
          height="400"
        />
        <source
          type="image/webp"
          media="(min-width:768px)"
          srcSet="
                images/location-tablet@1x.webp,
                images/location-tablet@2x.webp 2x
              "
          width="768"
          height="400"
        />
        <source
          type="image/webp"
          srcSet="
                images/location-mobile@1x.webp,
                images/location-mobile@2x.webp 2x
              "
          width="320"
          height="362"
        />
        <source
          type="image/png"
          media="(min-width:1280px)"
          srcSet="
                images/location-desktop@1x.png,
                images/location-desktop@2x.png 2x
              "
          width="1440"
          height="400"
        />
        <source
          type="image/png"
          media="(min-width:768px)"
          srcSet="
                images/location-tablet@1x.png,
                images/location-tablet@2x.png 2x
              "
          width="768"
          height="400"
        />
        <img
          src="images/location-mobile@1x.png"
          srcSet="images/location-mobile@2x.png 2x"
          alt="Местоположение офиса"
          className="map__image"
          width="320"
          height="362"
        />
      </picture>
      <iframe
        className="map__interactive-location"
        src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d1888.5454630863453!2d37.50522796070521!3d55.74336458360478!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b5495835c03ae5%3A0x6d69a3b0825233a5!2sSky%20Karting!5e0!3m2!1sru!2sru!4v1723552429572!5m2!1sru!2sru"
        height="560"
        style={{ border: 0 }}
        allowFullScreen
        title="Офис компании по адресу ул. Большая Конюшенная, д.19/8, Санкт-Петербург"
        loading="lazy"
        referrerPolicy="no-referrer-when-downgrade"
      ></iframe>
    </section>
  );
}

export default Map;
