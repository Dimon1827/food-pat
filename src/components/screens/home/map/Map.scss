@import '@/assets/styles/variables';

.map {
  position: relative;

  &--container {
    background-color: $gb-basic-grey;

    @media (min-width: $desktop-width) {
      background-color: $white-color;
    }
  }
}

.map__image {
  width: 100%;
  object-fit: cover;
}

.map__contacts {
  padding: 26px 0;

  &--color {
    background-color: $gb-basic-grey;

    @media (min-width: $desktop-width) {
      background-color: $white-color;
    }
  }

  @media (min-width: $tablet-width) {
    gap: 308px;
    padding: 61px 40px 59px;
  }

  @media (min-width: $desktop-width) {
    width: 565px;
    position: absolute;
    padding: 61px 80px;
    gap: 87px;
    left: 31px;
    top: 95px;
    z-index: 11;
  }
}

.map__interactive-location {
  display: block;
  border: none;
  width: 100%;
  max-width: 100%;
  position: absolute;
  overflow: hidden;
  top: 112px;
  z-index: 10;
  height: 362px;

  @media (min-width: $tablet-width) {
    height: 400px;
    top: 200px;
  }

  @media (min-width: $desktop-width) {
    top: 0;
  }
}
