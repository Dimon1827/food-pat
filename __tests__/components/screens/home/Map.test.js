import { render } from '@testing-library/react';
import Map from '@/components/screens/home/map/Map';

describe('Map', () => {
  it('Map renders correctly', () => {
    const { getByTestId } = render(<Map />);
    expect(getByTestId('map')).toMatchSnapshot();
  });
});
