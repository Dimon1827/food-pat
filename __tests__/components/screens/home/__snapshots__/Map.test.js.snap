// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Map renders correctly 1`] = `
<section
  class="map"
  data-testid="map"
>
  <div
    class="container"
  >
    <h2
      class="visually-hidden"
    >
      Контакты
    </h2>
    <div
      class="map__contacts map-contacts"
      data-testid="contacts"
    >
      <p
        class="map-contacts__text"
      >
        Наш центр можете отыскать по этому адресу
      </p>
      <div
        class="map-contacts__info"
      >
        <address
          class="map-contacts__address"
        >
          улица Багратионосовкая, д. 5
        </address>
        <address
          class="map-contacts__address"
        >
          Москва
        </address>
      </div>
    </div>
  </div>
  <picture>
    <source
      height="400"
      media="(min-width:1280px)"
      srcset="
                images/location-desktop@1x.webp,
                images/location-desktop@2x.webp 2x
              "
      type="image/webp"
      width="1440"
    />
    <source
      height="400"
      media="(min-width:768px)"
      srcset="
                images/location-tablet@1x.webp,
                images/location-tablet@2x.webp 2x
              "
      type="image/webp"
      width="768"
    />
    <source
      height="362"
      srcset="
                images/location-mobile@1x.webp,
                images/location-mobile@2x.webp 2x
              "
      type="image/webp"
      width="320"
    />
    <source
      height="400"
      media="(min-width:1280px)"
      srcset="
                images/location-desktop@1x.png,
                images/location-desktop@2x.png 2x
              "
      type="image/png"
      width="1440"
    />
    <source
      height="400"
      media="(min-width:768px)"
      srcset="
                images/location-tablet@1x.png,
                images/location-tablet@2x.png 2x
              "
      type="image/png"
      width="768"
    />
    <img
      alt=""
      class="map__image"
      height="362"
      src="images/location-mobile@1x.png"
      srcset="images/location-mobile@2x.png 2x"
      width="320"
    />
  </picture>
  <iframe
    allowfullscreen=""
    class="map__interactive-location"
    height="560"
    loading="lazy"
    referrerpolicy="no-referrer-when-downgrade"
    src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d1888.5454630863453!2d37.50522796070521!3d55.74336458360478!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b5495835c03ae5%3A0x6d69a3b0825233a5!2sSky%20Karting!5e0!3m2!1sru!2sru!4v1723552429572!5m2!1sru!2sru"
    style="border: 0px;"
    title="Офис компании по адресу ул. Большая Конюшенная, д.19/8, Санкт-Петербург"
  />
</section>
`;
